exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -174,15 +178,6 @@ func routeRecord(
 		}
 	}
 
-	dataStreamMode := datasetExists || namespaceExists
-	if !dataStreamMode {
-		prefix, prefixExists := getFromAttributes(indexPrefix, "", resourceAttr, scopeAttr, recordAttr)
-		suffix, suffixExists := getFromAttributes(indexSuffix, "", resourceAttr, scopeAttr, recordAttr)
-		if prefixExists || suffixExists {
-			return elasticsearch.Index{Index: fmt.Sprintf("%s%s%s", prefix, index, suffix)}, nil
-		}
-	}
-
 	// Receiver-based routing
```

Comment:
It seems that receiver-based routing has a higher precedence compared to the `data_stream.dataset` attribute, which isn't how it's documented.

Reply:
Nice catch. Fixed the bug and added a test: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38500/commits/5c9c85ab253f1c7b623c8e62d1960f526f0032a5